web/js/tasks.js

```diff
@@ -252,13 +252,45 @@ function removeFreshEmptyTask() {
     }
 }
 
+function getMinutes(time){
+    const hours = Number(time.split(':')[0]);
+    const minutes = Number(time.split(':')[1]);
+    return hours + minutes;
+}
+
+function checkTaskForOverlap(store){
+    let existingTasks = store.data.items.filter(x => !x.id.startsWith("ext"));
+    let unsavedTasks = store.data.items.filter(x => x.id.startsWith("ext"));
+
+    let overlapping = [];
+    let overlapsTasks = false;
+    let message = "";
+
+    existingTasks.forEach(x => {
+        unsavedTasks.forEach(t => {
+            if(getMinutes(t.data.endTime) > getMinutes(x.data.initTime) && getMinutes(t.data.initTime) < getMinutes(x.data.endTime)){
+                overlapsTasks = true;
+                overlapping.push(x);
+                message = "Task from " + x.data.initTime + " to " + x.data.endTime + " overlaps with task from " + t.data.initTime + " to " + t.data.endTime;
+            }
+            else if (getMinutes(t.data.initTime) == getMinutes(x.data.initTime) || getMinutes(t.data.endTime) == getMinutes(x.data.endTime)){
+                overlapsTasks = true;
+                overlapping.push(x);
+                message = "Task from " + x.data.initTime + " to " + x.data.endTime + " overlaps with task from " + t.data.initTime + " to " + t.data.endTime;
+            }
```

Comment:
I wonder if we can merge these blocks since they yield the same thing?

```suggestion
            if((getMinutes(t.data.endTime) > getMinutes(x.data.initTime) && getMinutes(t.data.initTime) < getMinutes(x.data.endTime)) || (getMinutes(t.data.initTime) == getMinutes(x.data.initTime) || getMinutes(t.data.endTime) == getMinutes(x.data.endTime))){
                overlapsTasks = true;
                overlapping.push(x);
                message = "Task from " + x.data.initTime + " to " + x.data.endTime + " overlaps with task from " + t.data.initTime + " to " + t.data.endTime;
            }
```

Maybe we can improve readability by storing the checks in variables:

```suggestion
            let overlapsTasksDuration = (getMinutes(t.data.endTime) > getMinutes(x.data.initTime) && getMinutes(t.data.initTime) < getMinutes(x.data.endTime));
            let coincidesInitOrEndTimes =  (getMinutes(t.data.initTime) == getMinutes(x.data.initTime) || getMinutes(t.data.endTime) == getMinutes(x.data.endTime));
            if (overlapsTasksDuration || coincidesInitOrEndTimes) {
                overlapsTasks = true;
                overlapping.push(x);
                message = "Task from " + x.data.initTime + " to " + x.data.endTime + " overlaps with task from " + t.data.initTime + " to " + t.data.endTime;
            }
```

Reply:
I noticed the message will be overwritten and show only the latest overlapping tasks, is it on purpose? It got me thinking if we want to show them all or only one, I don't have a strong opinion.

Reply:
Oops. Not on purpose. But I don't have a strong opinion since if there is more than one problem, a user will have to fix the first failure and simply get another failure for the next problem until they fix everything. I can try to get it to show all, since that was my original intent.

Reply:
The suggestions on readability are good. I like the idea of storing in variable to make it easier to parse.